### PR TITLE
[DOC] improve doc regarding events with 0 duration

### LINKED ISSUE #4562: [DOC] Better document how 0 duration events are handled
I think it might be nice to explicitly say how these 0 duration events are handled when creating a GLM but maybe that's because I'm not familiar with how it's done in other tools.

_Originally posted by @hugofluhr in https://github.com/nilearn/nilearn/issues/3884#issuecomment-2373645905_
            
places to check for improvement:
- design matrices construction
- plotting events
- associated examples

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4562 

Changes proposed in this pull request:

- update design matrix generation and plotting (and relevant examples) to mention how 0 duration events are handled

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
LGTM so far.

**Comment by Remi-Gau** [comment]
I don't think I will do more on this one.

File: examples/04_glm_first_level/plot_write_events_file.py [hunk 1]
```diff
@@ -146,10 +147,18 @@
...
+# Now lets plot the modulated and unmodulated events.
+fig = plot_event([events, modulated_events], figsize=(15, 5))
+fig.suptitle("Events and modulated events")
 show()
 
+# %%
+#
+# ..  note::
+#
+#       See how the modulation affects the height og the events.
```
Comment (Remi-Gau) [inline examples/04_glm_first_level/plot_write_events_file.py:159]:
```suggestion
#       See how the modulation affects the height of the events.
```

**Review by bthirion — APPROVED** [review]

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**